### ISSUE TITLE
woocommerce_applied_coupon

### ISSUE THREAD

**Issue by Manjupmf** [cedaro/woocommerce-coupon-links#23]
This plugins compliments another very useful plugin known as "[Cart links for WooCommerce](https://wordpress.org/plugins/soft79-cart-links-for-woocommerce/)".

On issue was that it was throwing an error "Something went wrong populating your cart:"
This error can be fixed by replacing 
"add_action( 'woocommerce_add_to_cart', 'cedaro_woocommerce_coupon_links', 30 );"  (in /woocommerce-coupon-links.php Line 71 ) with 
"add_action( 'woocommerce_applied_coupon', 'cedaro_woocommerce_coupon_links', 30 );"

**Comment by Manjupmf** [comment]
Hello @bradyvercher, firstly I must thank you for making the plugin available for the needy ones like me :) Alternative solutions were unnecessarily complex and expensive too.

I am using your plugin on my Woocommerce website store.pmfias.com. As mentioned in the comment above I have changed the add_action( 'woocommerce_add_to_cart', 'cedaro_woocommerce_coupon_links', 30 ); to add_action( 'woocommerce_applied_coupon', 'cedaro_woocommerce_coupon_links', 30 ); and plugin is working just excellent. I am using it in combination with "Cart links for WooCommerce". 

> "Is it only necessary for that particular plugin?"
Basically I want to first populate the cart with as many products as I want and then apply the coupon. "Cart links for WooCommerce" populates the cart and your plugin applies the coupon. 

Yes, the change I have suggested is required to make your plugin work with "Cart links for WooCommerce". Otherwise it was simply doing it's job fine. 

Manju.

**Comment by bradyvercher** [comment]
I ran a few tests with the Cart Links for WooCommerce plugin and it appeared to work with the current actions. Based on that error notice you're receiving, it looks like the actual notices may be masked by that plugin.

If you [take a look at the code](https://plugins.trac.wordpress.org/browser/soft79-cart-links-for-woocommerce/tags/1.1.4/soft79-cart-links-for-woocommerce.php#L213), if more than one notice is generated while populating the cart, that generic "Something went wrong populating your cart" message is displayed instead.

As far as I can tell, that's unrelated to this plugin, so I'm going to close this out, but if you can troubleshoot and provide more information that necessitates a change in this plugin, I'll be happy to revisit.

**Comment by Manjupmf** [comment]
No, thanks. The plugin works great. The conflict I mentioned could be due
to various other reasons as I have close to 30 plugins with 20 related to
woocommerce.
Keep up the good work.
